replication-api-impl/src/main/java/com/connexta/replication/api/impl/worker/Worker.java

```diff
@@ -0,0 +1,259 @@
+      CreateRequest request = new CreateRequestImpl(List.of(metadata));
+      boolean success = destination.createRequest(request);
+      if (!success) {
+        task.fail(ErrorCode.OPERATION_FAILURE,
+            String.format("Failed to %s metadata %s from %s to %s",
+                operation,
+                metadataId,
+                source.getSystemName(),
+                destination.getSystemName()));
+      }
+    } else {
+      LOGGER.info("Received task to create metadata on site {}, but it was deleted on {}",
+          destination.getSystemName(), destination.getSystemName());
+      task.complete();
+    }
+
+    ResourceInfo resourceInfo = task.getResource().orElse(null);
+    if (resourceInfo != null) {
+      URI uri = resourceInfo.getResourceUri().orElse(null);
+      if (uri != null) {
+        ResourceRequest request = new ResourceRequestImpl(metadata);
+        try {
+          ResourceResponse response = source.readResource(request);
+          Resource resource = response.getResource();
+          boolean success = destination
+              .createResource(new CreateStorageRequestImpl(List.of(resource)));
+          if (!success) {
+            task.fail(ErrorCode.OPERATION_FAILURE,
+                String.format("Failed to %s resource %s from %s to %s",
+                    operation,
+                    uri,
+                    source.getSystemName(),
+                    destination.getSystemName()));
+          }
+        } catch (AdapterException e) {
+          LOGGER.debug("Failed to retrieve resource", e);
+          task.fail(ErrorCode.OPERATION_FAILURE,
+              String.format("Failed to read resource from %s", destination.getSystemName()));
+        }
+      } else {
+        LOGGER.warn("No resource URI specified for task {}", task);
+        return;
+      }
+    }
+    task.complete();
+  }
+
+  private void doDelete(Task task, Metadata metadata) throws InterruptedException {
+    final String metadataId = task.getId();
+    final String operation = task.getOperation().name();
+    boolean success = destination.deleteRequest(new DeleteRequestImpl(List.of(metadata)));
+    if (!success) {
+      task.fail(ErrorCode.OPERATION_FAILURE,
+          String.format("Failed to %s metadata %s on %s",
+              operation,
+              metadataId,
+              destination.getSystemName()));
+    }
+    task.complete();
+  }
+
+  private boolean checkAdapterAvailabilities(Task task) throws InterruptedException {
+    try {
+      if (!local.isAvailable()) {
+        task.fail(ErrorCode.SITE_UNAVAILABLE,
+            String.format("Site %s is unavailable", local.getSystemName()));
+      }
+
+      if (!destination.isAvailable()) {
+        task.fail(ErrorCode.SITE_UNAVAILABLE,
+            String.format("Site %s is unavailable", destination.getSystemName()));
+      }
+    } catch (AdapterException e) {
+      LOGGER.debug("Error retrieving availability for adapters", e);
```

Comment:
❓ Shouldn't we handle those as failures of some sort?

Reply:
👍 Put it as an unknown error.